### PR TITLE
CC-4228: Configuration option to override expect-continue in upload protocol

### PR DESCRIPTION
**Pull request by wicknicks**
Signed-off-by: Arjun Satish <arjun@confluent.io>

### CODE REVIEW

File: kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkConnectorConfig.java [hunk 1]
```diff
@@ -427,6 +430,18 @@ public static ConfigDef newConfigDef() {
           "S3 Proxy Password"
       );
 
+      configDef.define(
+          S3_PROXY_USE_EXPECT_CONTINUE_CONFIG,
+          Type.BOOLEAN,
+          S3_PROXY_USE_EXPECT_CONTINUE_DEFAULT,
+          Importance.LOW,
+          "Sets if use expect continue should be enabled when accessing S3 through a proxy.",
```
Comment (kkonstantine) [inline kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkConnectorConfig.java]: This setting has implications. 

We might want to include info that is more related to this comment: 
https://github.com/aws/aws-sdk-java/blob/master/aws-java-sdk-core/src/main/java/com/amazonaws/ClientConfiguration.java#L326-L336

Rather what the javadoc for the method says. At least closer to what the first paragraph of this link ^^ says. 
Also, fyi, the http header is "expect continue". The word "use" is related more to the method in AWS SDK.

File: pom.xml [hunk 2]
```diff
@@ -62,6 +62,7 @@
     <properties>
         <confluent.maven.repo>http://packages.confluent.io/maven/</confluent.maven.repo>
         <licenses.version>5.0.4-SNAPSHOT</licenses.version>
+        <checkstyle.suppressions.location>checkstyle/suppressions.xml</checkstyle.suppressions.location>
```
[the remark this answers is not in the document]
Comment (wicknicks) [reply pom.xml]: This was necessary for getting the build to pass. Let me know if you want to handle this in a separate PR.

**Comment by wicknicks** [comment]
@kkonstantine addressed your comments. this is ready for a second pass.

File: kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkConnectorConfig.java [hunk 3]
```diff
@@ -427,6 +433,21 @@ public static ConfigDef newConfigDef() {
...
+          HEADERS_USE_EXPECT_CONTINUE_DEFAULT,
+          Importance.LOW,
+          "Enable/disable use of the HTTP/1.1 handshake using EXPECT: 100-CONTINUE during "
+              + "multi-part upload. If true, the client will wait for a 100 (CONTINUE) response "
+              + "before sending the request body. Else, the client uploads the entire request "
+              + "body without checking if the server is willing to accept the request.",
+          group,
+          ++orderInGroup,
+          Width.SHORT,
+          "S3 Proxy Use Expect Continue"
```
Comment (kkonstantine) [inline kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkConnectorConfig.java]: nit: I believe proxy is not relevant now. 
`S3 HTTP Send Uses Expect Continue`?

**Comment by wicknicks** [comment]
Thanks for the feedback, @kkonstantine! Merging this PR.

**Pull request merged**